### ISSUE TITLE
Parquet file owned by other user gets deleted upon dvc add

### ISSUE THREAD

**Issue by aernesto** [iterative/dvc#4348]
## Bug Report
 
On a Linux machine, in a folder `foo/` shared by various users, but not owned by me, I did:
```
git init .
dvc init .
git add .dvc
git commit -m "first commit"
```
The folder already had a file `bar.parquet` sitting in it, not owned by me either.
I did:
```
dvc add bar.parquet
```
And got the error
```
Adding...
ERROR: unexpected error - [Errno 1] Operation not permitted: '/baz/.dvc-cache/d0/3ea995971fc314c3087abbb3106e92.L6rCVNZWCFrzQRk53wdSCS'
```
 
**Most importantly**, the `foo/bar.parquet` file got deleted by the `dvc add` command above.
 
Note that I am using a shared `dvc` cache located at `/baz/.dvc-cache/`.
 
### Please provide information about your setup
 
**Output of `dvc version`:**
 
```console
$ dvc version
1.3.1
```
 
**Additional Information (if any):**
 
I could not reproduce the bug. After recreating `bar.parquet` manually, with me being the owner, I could run `dvc add bar.parquet` successfully.

**Comment by efiop** [comment]
Discord context: https://discordapp.com/channels/485586884165107732/485596304961962003/740974584437669930

**Comment by pared** [comment]
@aernesto the problem should be fixed in #4348
